lib/experimental/fonts/font-library/class-wp-rest-font-faces-controller.php

```diff
@@ -0,0 +1,461 @@
+
+	public function __construct() {
+		$post_type       = 'wp_font_face';
+		$this->post_type = $post_type;
+
+		$post_type_obj   = get_post_type_object( $post_type );
+		$this->rest_base = $post_type_obj->rest_base;
+
+		$parent_post_type       = 'wp_font_family';
+		$this->parent_post_type = $parent_post_type;
+		$parent_post_type_obj   = get_post_type_object( $parent_post_type );
+		$this->parent_base      = $parent_post_type_obj->rest_base;
+		$this->namespace        = $parent_post_type_obj->rest_namespace;
+	}
+
+	/**
+	 * Registers the routes for posts.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->parent_base . '/(?P<parent>[\d]+)/' . $this->rest_base,
+			array(
+				'args'        => array(
+					'parent' => array(
+						'description' => __( 'The ID for the parent font family of the font face.', 'gutenberg' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_font_faces_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'allow_batch' => $this->allow_batch,
+				'schema'      => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->parent_base . '/(?P<parent>[\d]+)/' . $this->rest_base . '/(?P<id>[\d]+)',
+			array(
+				'args'        => array(
+					'parent' => array(
+						'description' => __( 'The ID for the parent font family of the font face.', 'gutenberg' ),
+						'type'        => 'integer',
+					),
+					'id'     => array(
+						'description' => __( 'Unique identifier for the font face.', 'gutenberg' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_font_faces_permissions_check' ),
+					'args'                => array(),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+					'args'                => array(
+						'force' => array(
+							'type'        => 'boolean',
+							'default'     => false,
+							'description' => __( 'Whether to bypass Trash and force deletion.', 'default' ),
+						),
+					),
+				),
+				'allow_batch' => $this->allow_batch,
```

Comment:
Both `/font-families/` and `/font-faces/` API paths will accept files. Since the batch framework doesn't work with files. Could we delete this property and use the default, false?

Reply:
Good call. I know we previously discussed using the batch API for font faces, but it's not very useful without file uploads support. I'll remove this.